middleware/xray/segment.go

```diff
@@ -175,37 +200,41 @@ func (s *Segment) RecordResponse(resp *http.Response) {
 	s.HTTP.Response = responseData(resp)
 }
 
+// RecordContextResponse traces a context response if present in the context
+//
+// It sets Throttle, Fault, Error and HTTP.Response
+func (s *Segment) RecordContextResponse(ctx context.Context) {
+	resp := goa.ContextResponse(ctx)
+	if resp == nil {
+		return
+	}
+
+	s.Lock()
+	defer s.Unlock()
+
+	if s.HTTP == nil {
```

Comment:
maybe from this line on down could be factored out as a shared method that could also be reused by `RecordResponse`?
```go
func (s *Segment) recordResponse(status, respLength int)
```